frontend-react/src/pages/admin/AdminRevHistory.tsx

```diff
@@ -0,0 +1,273 @@
+import { Helmet } from "react-helmet";
+import React, { Suspense, useCallback, useState } from "react";
+import { Grid, GridContainer, Accordion } from "@trussworks/react-uswds";
+import { AccordionItemProps } from "@trussworks/react-uswds/lib/components/Accordion/Accordion";
+import { useParams } from "react-router-dom";
+
+import HipaaNotice from "../../components/HipaaNotice";
+import {
+    SettingRevision,
+    SettingRevisionParams,
+    useSettingRevisionEndpointsQuery,
+} from "../../network/api/Organizations/SettingRevisions";
+import Spinner from "../../components/Spinner";
+import { AuthElement } from "../../components/AuthElement";
+import { MemberType } from "../../hooks/UseOktaMemberships";
+import { jsonSortReplacer } from "../../utils/JsonSortReplacer";
+import { formatDate } from "../../utils/misc";
+import { StaticCompare } from "../../components/StaticCompare";
+
+/** Accordion components need data in specific format **/
+const dataToAccordionItems = (props: {
+    key: string; // used for React key and passed back to the onClickHandler
+    selectedKey: string;
+    onClickHandler: (
+        key: string,
+        itemClickedKey: string,
+        data: SettingRevision
+    ) => void;
+    data: SettingRevision[];
+}): AccordionItemProps[] => {
+    const results: AccordionItemProps[] = [];
+    if (props.data.length === 0) {
+        return [];
+    }
+
+    // should come back sorted by name and version from server. Sort by name, then by version
+    props.data.sort((a, b) =>
+        a.name === b.name ? a.version - b.version : a.name.localeCompare(b.name)
+    );
+
+    // group the data to make generating the prop content logic easier to follow
+    const grouped: { [key: string]: SettingRevision[] } = {};
+    props.data.forEach((item) => {
+        const name = item.name;
+        if (!grouped[name]) {
+            // key to array map. create initial entry for this group if needed
+            grouped[name] = [];
+        }
+        grouped[name].push(item);
+    });
+
+    // turn each group into a html list and add to the content of the accordion
+    for (let [key, settings] of Object.entries(grouped)) {
```

Comment:
can this be a reducer too? personal bias here, but feel like `for of` loops without a compelling reason are an antipattern

Reply:
`for (let [key, value] of Object.entries() {}` loops are an anti-pattern? I'd love to read more so I can stay up on the latest trends like this.

Personally, I tend to flip between array function (map, reduce, filter) operators and `for of` loops depending on complexity (or size) of work being done inside of them.

"reducers" can be fairly complex with their 4 params (e.g. what does previous mean on the first iteration... what's even being returned?)

There's also general confusing around how "reducers" work in React (`useReducer`) vs Native Js (`array.reduce`).

Reply:
it's a personal bias, but it's backed up by some other folks who have thought about these things: https://stackoverflow.com/questions/49420891/why-should-foreach-be-preferred-over-regular-iterators

beyond the point made here about mutability and side effects there's also

* confusion between `for of` and `for in`
* the fact that the iterating item value in the `for of` loop is a new instance of a variable, even though the loop syntax seems like it would work otherwise

i tend to only use `for of` if I'm dealing with a non-array iterable, but again, that's just me. Without a style guide I'm just a guy with an opinion

Reply:
But I _like_ your opinion, I'm just trying to understand/learn.